src/commands/config/create.ts

```diff
@@ -0,0 +1,180 @@
+import { create as CreatedWalletPath } from "../../implementations/wallet/create";
+import { getLogger } from "../../logger";
+import { highlight } from "../../utils";
+import { handler as CreateTemporaryDns } from "../dns/txt-record/create";
+import ConfigTemplate from "./config-template.json";
+import { CreateConfigCommand } from "./config.type";
+
+interface ConfigFile {
+  wallet: string;
+  forms: Form[];
+}
+
+interface Form {
+  defaults: {
+    issuers: (VerifiableDocumentIssuers | TransferableRecordIssuers)[];
+  };
+}
+interface VerifiableDocumentIssuers {
+  id?: string;
+  revocation?: { type: string };
+  name: string;
+  documentStore?: string;
+  identityProof: {
+    type: string;
+    location?: string;
+    key?: string;
+  };
+}
+interface TransferableRecordIssuers {
+  name: string;
+  tokenRegistry: string;
+  identityProof: {
+    type: string;
+    location?: string;
+  };
+}
+
+const { trace } = getLogger("config:create");
+
+export const command = "create [options]";
+
+export const describe = "Create a config file";
+
+export const builder = (yargs: Argv): Argv =>
+  yargs
+    .option("output-dir", {
+      alias: "od",
+      description: "Write output to a directory",
+      type: "string",
+      demandOption: true,
+    })
+    .option("encrypted-wallet-path", {
+      type: "string",
+      description: "Path to file containing wallet.json",
+      normalize: true,
+    })
+    .option("config-template-path", {
+      type: "string",
+      description: "Path to file containing config template",
+      normalize: true,
+      demandOption: true,
+    });
+
+export const handler = async (args: CreateConfigCommand): Promise<void> => {
+  trace(`Args: ${JSON.stringify(args, null, 2)}`);
+  if (!fs.existsSync(args.outputDir)) {
+    fs.mkdirSync(args.outputDir);
+  }
+
+  try {
+    let walletPath = "";
+    if (!args.encryptedWalletPath) {
+      info(`Please enter password to create wallet`);
+      const createWalletParams = {
+        outputFile: `${args.outputDir}/wallet.json`,
+        fund: "ropsten",
+      };
+      walletPath = await CreatedWalletPath(createWalletParams);
+      info(`Wallet created at ${walletPath}`);
+    }
+    const walletFilePath = walletPath ? walletPath : args.encryptedWalletPath;
+    const wallet = await readFile(walletFilePath);
+    const walletObject = JSON.parse(wallet);
+
+    const configFile: ConfigFile = ConfigTemplate;
+    configFile.wallet = wallet;
+
+    const template = JSON.parse(await readFile(args.configTemplatePath));
+    const formsInTemplate = template.forms;
+    let documentStoreAddress = "";
+    let tokenRegistryAddress = "";
+    let verifiableDocumentDnsName = "";
+    let tokenRegistryDnsName = "";
+    const typeOfDocuments = formsInTemplate.map((formTemplate: any) => formTemplate.type);
+    if (typeOfDocuments.includes("VERIFIABLE_DOCUMENT")) {
+      info(`Enter password to continue deployment of Document Store`);
```

Comment:
which password ? (same below)

Reply:
ah ok need to type again the password, it's a bit weird though

Reply:
yes the passwords are for deploying of document store and token registry. As we are just using the functions already created in existing OA-CLI.